pkg/controller/s3/bucket/CORSConfig.go

```diff
@@ -131,3 +143,34 @@ func (in *CORSConfigurationClient) Delete(ctx context.Context, bucket *v1beta1.B
 	).Send(ctx)
 	return awsclient.Wrap(err, corsDeleteFailed)
 }
+
+// LateInitialize does nothing because CORSConfiguration might have been deleted
+// by the user.
+func (in *CORSConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketCorsRequest(&awss3.GetBucketCorsInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		return awsclient.Wrap(resource.Ignore(s3.CORSConfigurationNotFound, err), corsGetFailed)
+	}
+
+	// We need the second check here because by default the CORS is not set
+	if external.GetBucketCorsOutput == nil || len(external.CORSRules) == 0 {
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()), "external output", external.GetBucketCorsOutput)
```

Comment:
I added the log statement here because I thought late-inits happen infrequently enough that it would not produce an excessive number of log statements. Also, since it's a debug log there shouldn't be any effect for (the average) end-user, right?